samples/bluetooth/peripheral_hids_keyboard/src/main.c

```diff
@@ -140,17 +156,38 @@ static struct keyboard_state {
 	u8_t keys_state[KEY_PRESS_MAX];
 } hid_keyboard_state;
 
+#if CONFIG_NFC_OOB_PAIRING
+static struct k_work adv_work;
+static struct bt_le_oob oob_local;
+
+static u8_t ndef_msg_buf[256]; /* Buffer used to hold an NFC NDEF message. */
+
 
+static void advertising_start(struct k_work *work)
```

Comment:
Please implement it as a separate function leaving the adevertising_start signature untouched. It is now hard to understand that in OOB this function does not change. External work function like ```advertising_start_work``` should be created and it may just call this ```advertising_start``` function.

Reply:
done